sdk/servicebus/service-bus/src/sender.ts

```diff
@@ -179,14 +181,19 @@ export class ServiceBusSenderImpl implements ServiceBusSender {
     const invalidTypeErrMsg =
       "Provided value for 'messages' must be of type ServiceBusMessage, ServiceBusMessageBatch or an array of type ServiceBusMessage.";
 
+    // link message span contexts
+    let spanContextsToLink: SpanContext[] = [];
+    if (isServiceBusMessage(messages)) {
+      messages = [messages];
+    }
+    let batch: ServiceBusMessageBatch;
     if (Array.isArray(messages)) {
-      const batch = await this.createBatch(options);
-
-      for (const message of messages) {
+      batch = await this.createBatch(options);
+      for (let message of messages) {
```

Comment:
Reason for moving from const to let in the loop?

Reply:
Good Catch!!
Ah.. Switched this when I was following the event-hubs way of instrumenting the messages (duplicated for arrays and tryAdd).

I had later updated the arrays path to reuse the tryAdd options and missed this "const" to "let".

Reply:
Updated